bin/dnf-automatic.in

```diff
@@ -23,7 +23,11 @@ import os
 import sys
 
 here = sys.path[0]
-if here != '/usr/bin':
+if here == '/usr/bin':
+    # we never import Python modules from /usr/bin
+    # removing this lowers the risk of accidental imports of weird files
+    del sys.path[0]
```

Comment:
This apparently works but for some reason makes me cringe. Would you change it to `sys.path.pop(0)`?

Reply:
I consider pop semantically wrong here, as I don't need to retrieve the value. However, if this would be the only blocker, I can definitively change it.

Reply:
Well I don't really know what's the canonical way in Python to remove an item from the list, but the `del` statement doesn't fit syntactically and to me is very unintuitive in what's it doing with the list. I've done a quick google and I haven't found what's the canonical way but this stands out as weird to me.

Wrt. `pop()`: A better solution to the `del` statement would be to have a method called e.g. `delete()` that'd delete an item by index. But then the functionality would be the same as `pop()` except there's no returned value, hence the `delete()` method would actually be redundant. That's my justification for why I think `pop()` is the method to use and `del` just seems a weird historical syntax.

Not a blocker though. I'd still prefer `pop()` but if you still insist, keep the `del` :slightly_smiling_face:

Reply:
Standard Python containers don't have `.delete()` methods because `del` is a statement :)

Reply:
As a Python core developer, I can say that ``del sys.path[0]`` is the good way (and the most command method) to delete the first item of a list.